### DISCUSSION TITLE
ABLStats in prescribed mode?

### DISCUSSION THREAD

**Discussion by gantech** [Exawind/amr-wind#401] [Q&A]
Is there an ABLStats class for "prescribed" statistics for an inflow/outflow simulation? https://github.com/Exawind/amr-wind/blob/main/amr-wind/wind_energy/ABL.cpp#L26 

Is anyone working on this and has a branch in progress?

**Comment by sayerhs** [comment]
@gantech @shashankNREL and I had a discussion about _unifying_ ABL stats for inflow/outflow (from _precursor_) as well as WRF-style forcing. Check with him on what he has done so far. 

A while ago, I had started refactoring functionality in `ABLStats` class into `ABLStatsBase` (which is what is used by `ABL` class now), with the ability to specify the type of Stats/profile data it supports https://github.com/Exawind/amr-wind/blob/ba0455d93d8b905ae521e5b939c333649889eed8/amr-wind/wind_energy/ABLStatsBase.H#L15-L18

To do this cleanly, the `FieldPlaneAveraging` classes also need to be refactored a bit. They should not hold the data (profiles) instead only compute them given a `struct` (e.g., `ABLState`) that holds the data. This way you can _decouple_ the current dependence of ABL (and its physics implementation) on the `PlaneAveraging` classes. Instead it should just obtain an `ABLState` struct that holds all information necessary for interpolation.
